src/graph/interface/op_schema.cpp

```diff
@@ -102,15 +102,23 @@ op_schema_t &op_schema_t::set_output(
     return *this;
 }
 
-op_schema_t &op_schema_t::set_commutative_inputs() {
-    assertm(num_inputs_.size() == 1 && *(num_inputs_.begin()) == 2,
-            "commutative inputs can only be enabled for ops with two inputs");
+op_schema_t &op_schema_t::set_commutative_inputs(
+        const std::set<size_t> &inputs) {
+    assertm(num_inputs_.size() == 1 && *(num_inputs_.begin()) > 1,
+            "The number of commutative inputs is at least two");
+    assertm(std::all_of(inputs.begin(), inputs.end(),
+                    [&](size_t idx) { return idx < *(num_inputs_.begin()); }),
+            "each input value must be less than input numbers");
     commutative_inputs_enabled_ = true;
+    commutative_inputs_ = inputs;
     return *this;
 }
 
-bool op_schema_t::get_commutative_inputs() const {
-    return commutative_inputs_enabled_;
+bool op_schema_t::get_commutative_inputs(const std::set<size_t> &inputs) const {
+
+    return commutative_inputs_enabled_
+            && std::includes(commutative_inputs_.begin(),
+                    commutative_inputs_.end(), inputs.begin(), inputs.end());
```

Comment:
Not sure how this function is used. But from it's implementation, i think the function name can be improved:

```
bool op_schema_t::is_commutative_inputs(size_t input1, size_t input2) {
      // make sure both {input1, input2} and {input2, input1} are checked.
      const bool ret = .....;
      return ret;
}
```

Reply:
In fact, the inputs may exceed two, and while we currently don't have such an op, it might be beneficial to retain this extensibility for future use.

Reply:
In actual usage, one way this function is called is with no parameters, solely to check whether the current op supports commutativity. If we change the parameters to `size_t input1, size_t input2`, it would make this zero-argument usage less straightforward or even impossible without overloading or adding a separate method.

Reply:
My point is not the arguments of the function. It's more about the function name. You can still use `std::set` to pass the inputs.

- is_commutative_inputs(inputs): check if the given inputs are commutative, return true or false accordingly.
- get_commutative_inputs(): return the set of commutative inputs.

Reply:
The function name has been changed to `is_commutative_inputs`

Reply:
Since the zero-argument usage is separated, the arguments have been changed to `const size_t input0, const size_t input1`